### PR TITLE
Initial acceptance testing setup.

### PR DESCRIPTION
**Pull request by fotinakis**
Added:
- `tests/acceptance` folder
- `test-acceptance` Makefile endpoint, and modified `test-python` to avoid running acceptance tests until you're ready to integrate that into your full testing workflow.
- `LiveServerTestCase` utilities

@dcramer

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -102,7 +102,12 @@ test-js:
 
 test-python:
 	@echo "--> Running Python tests"
-	py.test tests || exit 1
+	py.test tests/integration tests/sentry || exit 1
+	@echo ""
+
+test-acceptance:
```
Comment (mattrobenolt) [inline Makefile]: `test-acceptance` needs to get added into `.PHONY`

Comment (fotinakis) [inline Makefile]: Done.

**Comment by mattrobenolt** [comment]
When I try to run this, I get the following:

```
Exception AttributeError: "'Service' object has no attribute 'process'" in <bound method Service.__del__ of <selenium.webdriver.phantomjs.service.Service object at 0x10c3c9410>> ignored
```

Should there be more involved to get this setup as well?

**Comment by mattrobenolt** [comment]
Also:

```
WebDriverException: Message: 'phantomjs' executable needs to be in PATH.
```

**Comment by fotinakis** [comment]
Per discussion, going to add phantomjs as an npm dependency and then try to point selenium at that binary.

**Comment by fotinakis** [comment]
Ok @mattrobenolt, try again?

**Comment by mattrobenolt** [comment]
Now I get something new:

```
$ py.test tests/acceptance
================================== test session starts ===================================
platform darwin -- Python 2.7.11 -- py-1.4.26 -- pytest-2.6.4
plugins: cov, django, timeout, xdist
collected 1 items

tests/acceptance/test_static.py F

======================================== FAILURES ========================================
____________________________ StaticAcceptanceTest.test_static ____________________________
Traceback (most recent call last):
  File "/Users/matt/code/getsentry/sentry/tests/acceptance/test_static.py", line 6, in test_static
    self.browser.get(self.live_server_url)
  File "/Users/matt/.virtualenvs/sentry/lib/python2.7/site-packages/selenium/webdriver/remote/webdriver.py", line 245, in get
    self.execute(Command.GET, {'url': url})
  File "/Users/matt/.virtualenvs/sentry/lib/python2.7/site-packages/selenium/webdriver/remote/webdriver.py", line 231, in execute
    response = self.command_executor.execute(driver_command, params)
  File "/Users/matt/.virtualenvs/sentry/lib/python2.7/site-packages/selenium/webdriver/remote/remote_connection.py", line 395, in execute
    return self._request(command_info[0], url, body=data)
  File "/Users/matt/.virtualenvs/sentry/lib/python2.7/site-packages/selenium/webdriver/remote/remote_connection.py", line 463, in _request
    resp = opener.open(request, timeout=self._timeout)
  File "/usr/local/Cellar/python/2.7.11/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 431, in open
    response = self._open(req, data)
  File "/usr/local/Cellar/python/2.7.11/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 449, in _open
    '_open', req)
  File "/usr/local/Cellar/python/2.7.11/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 409, in _call_chain
    result = func(*args)
  File "/usr/local/Cellar/python/2.7.11/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 1227, in http_open
    return self.do_open(httplib.HTTPConnection, req)
  File "/usr/local/Cellar/python/2.7.11/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 1200, in do_open
    r = h.getresponse(buffering=True)
  File "/usr/local/Cellar/python/2.7.11/Frameworks/Python.framework/Versions/2.7/lib/python2.7/httplib.py", line 1136, in getresponse
    response.begin()
  File "/usr/local/Cellar/python/2.7.11/Frameworks/Python.framework/Versions/2.7/lib/python2.7/httplib.py", line 453, in begin
    version, status, reason = self._read_status()
  File "/usr/local/Cellar/python/2.7.11/Frameworks/Python.framework/Versions/2.7/lib/python2.7/httplib.py", line 409, in _read_status
    line = self.fp.readline(_MAXLINE + 1)
  File "/usr/local/Cellar/python/2.7.11/Frameworks/Python.framework/Versions/2.7/lib/python2.7/socket.py", line 480, in readline
    data = self._sock.recv(self._rbufsize)
timeout: timed out
---------------------------------- Captured stdout call ----------------------------------
----------------------------------------
Exception happened during processing of request from ('127.0.0.1', 56910)
----------------------------------------
---------------------------------- Captured stderr call ----------------------------------
Traceback (most recent call last):
  File "/usr/local/Cellar/python/2.7.11/Frameworks/Python.framework/Versions/2.7/lib/python2.7/wsgiref/handlers.py", line 86, in run
    self.finish_response()
  File "/usr/local/Cellar/python/2.7.11/Frameworks/Python.framework/Versions/2.7/lib/python2.7/wsgiref/handlers.py", line 128, in finish_response
    self.write(data)
  File "/usr/local/Cellar/python/2.7.11/Frameworks/Python.framework/Versions/2.7/lib/python2.7/wsgiref/handlers.py", line 212, in write
    self.send_headers()
  File "/usr/local/Cellar/python/2.7.11/Frameworks/Python.framework/Versions/2.7/lib/python2.7/wsgiref/handlers.py", line 271, in send_headers
    self._write(str(self.headers))
  File "/usr/local/Cellar/python/2.7.11/Frameworks/Python.framework/Versions/2.7/lib/python2.7/socket.py", line 328, in write
    self.flush()
  File "/usr/local/Cellar/python/2.7.11/Frameworks/Python.framework/Versions/2.7/lib/python2.7/socket.py", line 307, in flush
    self._sock.sendall(view[write_offset:write_offset+buffer_size])
error: [Errno 32] Broken pipe
Traceback (most recent call last):
  File "/Users/matt/.virtualenvs/sentry/lib/python2.7/site-packages/django/test/testcases.py", line 1035, in _handle_request_noblock
    self.process_request(request, client_address)
  File "/usr/local/Cellar/python/2.7.11/Frameworks/Python.framework/Versions/2.7/lib/python2.7/SocketServer.py", line 321, in process_request
    self.finish_request(request, client_address)
  File "/usr/local/Cellar/python/2.7.11/Frameworks/Python.framework/Versions/2.7/lib/python2.7/SocketServer.py", line 334, in finish_request
    self.RequestHandlerClass(request, client_address, self)
  File "/Users/matt/.virtualenvs/sentry/lib/python2.7/site-packages/django/core/servers/basehttp.py", line 126, in __init__
    super(WSGIRequestHandler, self).__init__(*args, **kwargs)
  File "/usr/local/Cellar/python/2.7.11/Frameworks/Python.framework/Versions/2.7/lib/python2.7/SocketServer.py", line 657, in __init__
    self.finish()
  File "/usr/local/Cellar/python/2.7.11/Frameworks/Python.framework/Versions/2.7/lib/python2.7/SocketServer.py", line 716, in finish
    self.wfile.close()
  File "/usr/local/Cellar/python/2.7.11/Frameworks/Python.framework/Versions/2.7/lib/python2.7/socket.py", line 283, in close
    self.flush()
  File "/usr/local/Cellar/python/2.7.11/Frameworks/Python.framework/Versions/2.7/lib/python2.7/socket.py", line 307, in flush
    self._sock.sendall(view[write_offset:write_offset+buffer_size])
error: [Errno 32] Broken pipe
=============================== 1 failed in 12.62 seconds ================================
```

**Comment by mattrobenolt** [comment]
You also need to modify the `test-python-coverage` make target so it doesn't try to run these with normal tests, like you did for `test-python`. This is what gets run by Travis.

**Comment by fotinakis** [comment]
@mattrobenolt fixed the phantomjs zombie problem, it turns out to be a selenium bug so we have to unfortunately send a manual SIGTERM for now.

**Comment by mattrobenolt** [comment]
> it turns out to be a selenium bug so we have to unfortunately send a manual SIGTERM for now.

lol

Assuming tests pass, we can merge this.

**Comment by fotinakis** [comment]
Good to go here @mattrobenolt, thanks for your help today.

**Comment by fotinakis** [comment]
@mattrobenolt good to merge here?

**Comment by mattrobenolt** [comment]
Likely, it'll be worth noting that now simply calling `$ py.test tests` or `$ py.test` will cause these tests to be run as well, which I assume is something that we'd need to be aware of since it's unlikely something we want to be running ourselves locally.

Not sure if we want to move these outside normal `tests` directory? Or we can tell `py.test` to ignore this directory.

@getsentry/infrastructure

**Comment by dcramer** [comment]
@mattrobenolt people can say py.test tests/sentry. There is no reason to have multiple tests directories, or to special case one type of test.

**Comment by fotinakis** [comment]
Yup, happy to do whatever you decide here. By default Percy will be disabled locally, so you won't have to worry about that part when running `py.test` manually, but you would running a set of potentially slow tests.

**Comment by mattrobenolt** [comment]
> people can say py.test tests/sentry.

Yeah, that's fine. Just noting change in behavior here. :)

Everything looks good. :cookie: thanks for this!

**Comment by fotinakis** [comment]
Thanks!

**Pull request merged**